thirdeye/thirdeye-pinot/src/main/java/org/apache/pinot/thirdeye/detection/components/PercentageChangeRuleDetector.java

```diff
@@ -76,16 +80,34 @@
 
   @Override
   public DetectionResult runDetection(Interval window, String metricUrn) {
+    long startMillis = window.getStartMillis();
+
+    // align start day to the user specified week start
+    if (Objects.nonNull(this.weekStart)) {
+      startMillis = window.getStart().withDayOfWeek(weekStart.getValue()).minusWeeks(1).getMillis();
```

Comment:
The startMills calculation should be done when creating the task. Why calculating this again?

Reply:
Yes. This is to ensure that the window is aligned to the specified boundary because in preview it can be adjusted to any day.